### PR TITLE
Link foldM docs to purescript-safely.

### PR DESCRIPTION
**Pull request by matthewleon**

### CODE REVIEW

File: src/Data/Foldable.purs [hunk 1]
```diff
@@ -177,6 +177,8 @@ fold = foldMap id
 -- |
 -- | Note: this function is not generally stack-safe, e.g., for monads which 
 -- | build up thunks a la `Eff`.
+-- | For a stack-safe alternative, see
```
Comment (paf31) [inline src/Data/Foldable.purs]: Might want to link to the actual function docs, and also note that it only works for `List`?

Comment (matthewleon) [reply src/Data/Foldable.purs]: Good point. Will update.

File: src/Data/Foldable.purs [hunk 2]
```diff
@@ -177,6 +177,8 @@ fold = foldMap id
 -- |
 -- | Note: this function is not generally stack-safe, e.g., for monads which 
 -- | build up thunks a la `Eff`.
+-- | For a stack-safe alternative that works on lists, see
+-- | [purescript-safely](https://pursuit.purescript.org/packages/purescript-safely/3.0.0/docs/Control.Safely#v:foldM)
```
Comment (JordanMartinez) [inline src/Data/Foldable.purs:181]: This link will become outdated if `purescript-safely` ever makes a new release (like now during v0.14.0 migration phase).

Could we instruct people to search for `foldM` in the `purescript-safely` package instead?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]

**Comment by paf31** [comment]
`safely` is pretty slow, and I'm unlikely to keep it up to date anyway.

In any case, for a better approach, I'd recommend considering adding a `Dissectible` subclass of `Traversable` instead: https://blog.functorial.com/posts/2017-06-18-Stack-Safe-Traversals-via-Dissection.html

**Comment by hdgarrood** [comment]
Interesting, thanks for the link!

**Comment by JordanMartinez** [comment]
Where should the Dissectible code go? In this repo or a different one?

**Comment by hdgarrood** [comment]
I’d prefer to put it in a different repo at first.

**Comment by JordanMartinez** [comment]
Similar to how we've handled other changes? Meaning, put it outside 'core' first and only add it in once it's proven itself?

I'd agree with that. Regardless, I believe we can close this PR.

**Comment by hdgarrood** [comment]
Yes, I think so - it’s probably best to avoid linking to things which are likely to go out of date.

**Comment by hdgarrood** [comment]
Thanks anyway for the PR @matthewleon!

**Pull request closed without merging**